### PR TITLE
Add hashicorp packer configuration for AMI creation

### PR DESCRIPTION
**Pull request by ddub**
Configuration updates:
Only set SERVER_NAME and GITHUB_PASSWORD if supplied
Default behaviour will now be to respond to all HTTP hosts and only use
password authentication for github is explicitly set

TODO:
- Add (terraform) provisioning example
- Cluster multiple instances together

### CODE REVIEW

**Comment by markstory** [comment]
Looks great so far!

File: README.mdown [hunk 1]
```diff
@@ -53,6 +53,17 @@ php composer.phar install
 
 Once the dependencies are installed you should configure the repositories you want to review.
 
+## Using AWS
+
+To use aws, you can use [packer](http://packer.io) to generate an Amazon Machine Image to run the application
```
Comment (adrianmoisey) [inline README.mdown]: What about using `https://www.packer.io/`? That way there's no redirect for anyone that clicks on that.

File: packer.json [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      "source": "logging.ini",
+      "destination": "logging.ini"
+    },
+    {
+      "type": "shell",
+      "execute_command": "sudo -E sh -e '{{ .Path }}'",
+      "inline": [
+        "DEBIAN_FRONTEND=noninteractive apt-get update",
+        "DEBIAN_FRONTEND=noninteractive apt-get upgrade -y",
+        "DEBIAN_FRONTEND=noninteractive apt-get update",
```
Comment (adrianmoisey) [inline packer.json]: This happens 2 lines up. Is it intentional?

Comment (ddub) [inline packer.json]: yeah, the apt-get install fails without the update after the upgrade, and the update doesn't work is the update doesn't happen first :-/

Comment (adrianmoisey) [inline packer.json]: That sounds broken. I've never had this behaviour with apt-get before. I wonder if Packer is doing something strange.

**Pull request merged**